staging/src/k8s.io/kubectl/pkg/cmd/scale/scale.go

```diff
@@ -234,7 +230,7 @@ func (o *ScaleOptions) RunScale() error {
 
 	var waitForReplicas *scale.RetryParams
 	if o.Timeout != 0 && o.dryRunStrategy == cmdutil.DryRunNone {
-		waitForReplicas = scale.NewRetryParams(1*time.Second, timeout)
+		waitForReplicas = scale.NewRetryParams(1*time.Second, o.Timeout)
```

Comment:
what is the difference between this and the retry params that still use 5 minutes? should we default o.Timeout to 5 minutes (rather than 0 meaning "no timeout") to preserve current behavior by default?

Reply:
Thanks for your review @liggitt !

Before #63198 merged, the code like this:
```go
cmd.Flags().DurationVar(&o.Duration, "timeout", o.Duration, "The length of time to wait before giving up on a scale operation, zero means don't wait. Any other values should contain a corresponding time unit (e.g. 1s, 2m, 3h).")
retry := kubectl.NewRetryParams(kubectl.Interval, kubectl.Timeout)
var waitForReplicas *kubectl.RetryParams
if timeout := o.Duration; timeout != 0 {
    waitForReplicas = kubectl.NewRetryParams(kubectl.Interval, timeout)
}
```
Notice that the `timeout` of `waitForReplicas = kubectl.NewRetryParams(kubectl.Interval, timeout)` is equal to `o.Duration` !

After #63198, we rename `o.Duration` as `o.Timeout`, but its meaning hasn't changed. 

```go
cmd.Flags().DurationVar(&o.Timeout, "timeout", 0, "The length of time to wait before giving up on a scale operation, zero means don't wait. Any other values should contain a corresponding time unit (e.g. 1s, 2m, 3h).")
retry := kubectl.NewRetryParams(kubectl.Interval, kubectl.Timeout)
var waitForReplicas *kubectl.RetryParams
if o.Timeout != 0 {
    waitForReplicas = kubectl.NewRetryParams(kubectl.Interval, timeout)
}
```

However, we confused global `timeout` with local `timeout`, the `timeout` of `waitForReplicas = kubectl.NewRetryParams(kubectl.Interval, timeout)` is not equal to `o.Timeout`. Obviously, we should use `o.Timoeout` here for consistency!

If I understand it correctly, the default value zero means do not wait for condition `scaleHasDesiredReplicas` is satisfied rather than "no timeout". This behavior is consistent with the original behavior before #63198. 

If I misunderstand, please point out @liggitt 😄

Reply:
@SataQiu honestly, we could make both configurable, I don't see any objections in that. Feel free to open a followup.